build.psm1

```diff
@@ -2485,19 +2485,21 @@ function Start-PSBootstrap {
             Write-LogGroupEnd -Title "Install Windows Dependencies"
         }
 
-        if ($Scenario -in 'All', 'Tools') {
-            Write-LogGroupStart -Title "Install .NET Global Tools"
-            Write-Log -message "Installing .NET global tools"
+        # Ensure dotnet is available
+        Find-Dotnet
 
-            # Ensure dotnet is available
-            Find-Dotnet
+        if (-not $env:TF_BUILD) {
+            if ($Scenario -eq 'DotNet' -or $Scenario -eq 'Both') {
```

Comment:
This is wrong. The `All` is a more recent change in master. The `Both` value in release branch is outdated. So, you should keep the master code as is, and only add the `if (-not $env:TF_BUILD) {` and move `Find-Dotnet` as the ADO PRs did.